apps/emqx/src/emqx_shared_sub.erl

```diff
@@ -315,7 +317,10 @@ do_pick_subscriber(Group, Topic, round_robin, _ClientId, _SourceTopic, Count) ->
             N -> (N + 1) rem Count
         end,
     _ = erlang:put({shared_sub_round_robin, Group, Topic}, Rem),
-    Rem + 1.
+    Rem + 1;
+do_pick_subscriber(Group, Topic, round_robin_per_node, _ClientId, _SourceTopic, Count) ->
+    %% reset the counter back to zero if counter > subscriber count to avoid the counter grow larger than the current subscriber count
+    ets:update_counter(?SHARED_SUBS_ROUND_ROBIN_COUNTER, {Group, Topic}, {2, 1, Count, 1}).
```

Comment:
if configuration is changed at runtime - lets say from round_robin to round_robin_per_node the process will crash here. Is this something that must be handled or is changing configuration at runtime not a use-case?

Reply:
Hot reloading of the configuration is necessary. You should use ets:update_counter/4 here